stdlib/public/core/Builtin.swift

```diff
@@ -307,6 +301,17 @@ func _uncheckedUnsafeAssume(_ condition: Bool) {
   _ = Builtin.assume_Int1(condition._value)
 }
 
+// This function is no longer used but must be kept for ABI compatibility
+// because references to it may have been inlined.
+@usableFromInline
+internal func _branchHint(_ actual: Bool, expected: Bool) -> Bool {
+  // The LLVM intrinsic underlying int_expect_Int1 now requires an immediate
+  // argument for the expected value so we cannot call it here. This should
+  // never be called in cases where performance matters, so just return the
+  // value without any branch hint.
+  return actual
+}
```

Comment:
I think it would be nice to introduce a `Swift50ABICompatibility.swift` and move this there so that we know what version we are maintaining compatibility with.  But, that is beyond the scope of this change.

Reply:
+1. My other PRs pending SE introduce a [LegacyABI.swift](https://github.com/apple/swift/pull/22869/files#diff-7a51bc96e3ab3b77dbdaf22266fe7a4e) file. It doesn't matter to me what's it's called, but splitting them out improves maintainability.